cognitedata/reveal
# improvement: Loading 360 descriptors for large sets no longer freeze the application

#### Type of change

![Refactor](https://img.shields.io/badge/Type-Refactor-lightgrey) 

#### Jira ticket :blue_book:


https://cognitedata.atlassian.net/browse/REV-789

## Description :pencil:
I found that the problem was in Cdf360ImageEventProvider and mergeDescriptors where we go through all files (in this case 43284) for each event (7222) which is a really big loop. 

Files are now represented by a Map so that lookup can be done in mergeDescriptors. I also replaced autoPagingToArray with autoPagingEach so that the files can be inserted directly into the map as they arrive.

## How has this been tested? :mag:
Using the Performance and Network tab in the debug window. Since "Performance" gave some weird results I also added timestamps in code and printed out how long different parts of the code took to finish.

Tested mostly on Windows/Chrome, but I did profile once on Android/Chrome.

## Test instructions :information_source:
1. Open any CDF model.
2. In the GUI open "360 Images" and add "Hibernia_RS2" as site_id in options.
3. Click "Add image set". (This can be done several times without refreshing the browser, but performance is affected by caching, so disable this in the Network-tab for consistent results.).
4. Move the camera around and see that there is no freezing going on 🧊 .

Optional: Use the Performance and Network tab to profile the loading.

## Checklist :ballot_box_with_check:
